### PR TITLE
Protect segment from being released while query/search

### PR DESCRIPTION
**Pull request by yah01**
/kind bug
related #26249

### CODE REVIEW

File: internal/querynodev2/segments/segment.go [hunk 1]
```diff
@@ -199,6 +199,16 @@ func (s *LocalSegment) isValid() bool {
 	return s.ptr != nil
 }
 
+// Provide ONLY the read lock operations,
+// don't make `mut` public to avoid abusing of the mutex
+func (s *LocalSegment) RLock() {
+	s.mut.RLock()
```
Comment (xiaofan-luan) [inline internal/querynodev2/segments/segment.go:205]: rename it as closeLock

Comment (xiaofan-luan) [inline internal/querynodev2/segments/segment.go:205]: after RLock,
need to check if segment.ptr == nil?
what happens if we get the lock but delete Segment alread done?

Comment (yah01) [reply internal/querynodev2/segments/segment.go:205]: This protects the whole segmen, not for the `ptr` only, keep it?

Comment (yah01) [reply internal/querynodev2/segments/segment.go:205]: It should fail if got a `nil` pointer, will fix this

Comment (xiaofan-luan) [reply internal/querynodev2/segments/segment.go:205]: name it to some better name~

**Pull request merged**